src/python_testing/TC_BINFO_3_2.py

```diff
@@ -0,0 +1,112 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import json
+import logging
+from time import sleep
+from typing import Any
+
+import chip.clusters as Clusters
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_BINFO_3_2(MatterBaseTest):
+    async def read_binfo_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.BasicInformation
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_BINFO_3_2(self) -> str:
+        return "[TC-BOOLCFG-3.2] Attributes with DUT as Server"
+
+    def steps_TC_BINFO_3_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "TH reads ConfigurationVersion from the DUT and stores the value as initialConfigurationVersion",
+                     "Verify that the value is in the inclusive range of 1 to 4294967295"),
+            TestStep(3, "Change the configuration version in a way which results in functionality to be added or removed (e.g. rewire thermostat to support a new mode)"),
+            TestStep(4, "TH reads ConfigurationVersion from the DUT",
+                     "Verify that the value is higher than the value of initialConfigurationVersion"),
+        ]
+        return steps
+
+    def pics_TC_BINFO_3_2(self) -> list[str]:
+        pics = [
+            "BINFO.S",
+            "BINFO.S.M.DeviceConfigurationChange",
+        ]
+        return pics
+
+    def _send_named_pipe_command(self, command_dict: dict[str, Any]):
```

Comment:
crossing the PR streams here...https://github.com/project-chip/connectedhomeip/pull/37637. Don't need to do anything in this PR, BTW, just close this, but figured it was worth mentioning for the future since you tend to write lots of tests.

Reply:
Funny, I actually thought it was strange that we had this duplicated code in the tests, so nice to see it is being sorted. I guess the path forward is to update this test in what ever order the two PRs land. Thanks for sharing.